ivy-test.el

```diff
@@ -939,6 +939,23 @@ a buffer visiting a file."
              "C-M-j")))
     (ivy-mode ivy-mode-reset-arg)))
 
+(ert-deftest ivy-starts-with-dotslash ()
+  (should
+   (equal
+    '((ok) (ok))
+    (delete
+     nil
+     (mapcar
+      (lambda (p)
+        (when (ivy--starts-with-dotslash (car p)) (list (cdr p))))
+      (list
+       (cons "./test1" 'ok)
+       (cons ".\\test2" 'ok)
+       (cons "test3" 'fail)
+       (cons "t/est4" 'fail)
+       (cons "t\\est5" 'fail)
+       (cons "tes./t6" 'fail)))))))
```

Comment:
Nitpick: Strings and symbols are self-quoting, and we're not modifying the alist, so a literal `'(("..." . ok) ...)` would suffice here.

Having said that, I think this test can be written in a more straight-forward way:

```el
(mapc (lambda (p)
        (should (eq (cdr p) (and (ivy--starts-with-dotslash (car p)) t))))
      '(("./test1" . t)
        (".\\test2" . t)
        ("test3")
        ("t/est4")
        ("t\\est5")
        ("tes./t6")))
```

or similar. Having said that, it is more idiomatic to keep logic to a minimum in ERT tests:

```el
(should (ivy--starts-with-dotslash "./test1"))
(should (ivy--starts-with-dotslash ".\\test2"))
(should (not (ivy--starts-with-dotslash "test3")))
(should (not (ivy--starts-with-dotslash "t/est4")))
(should (not (ivy--starts-with-dotslash "t\\est5")))
(should (not (ivy--starts-with-dotslash "tes./t6")))
```

Although I'm not usually a fan of repeating myself, tests are an exception where the latter style is instantly more obvious and possibly produces cleaner backtraces (though I'm not sure about this last point).

Reply:
I did not really pay much attention to the way these were written to be honest.
I like the last option more.